internal/driver/driver.go

```diff
@@ -81,21 +93,98 @@ func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModel.As
 		return errors.NewCommonEdgeXWrapper(err)
 	}
 
-	if err := ds.ListenForCustomConfigChanges(&d.serviceConfig.OPCUAServer.Writable, WritableInfoSectionName, d.updateWritableConfig); err != nil {
-		return errors.NewCommonEdgeX(errors.Kind(err), fmt.Sprintf("unable to listen for changes for '%s' custom configuration", WritableInfoSectionName), err)
+	// Add listener for username and password changes in insecure secrets config section
+	if err := ds.ListenForCustomConfigChanges(&d.serviceConfig.OPCUAServer.Writable, InsecureSecretsConfigSectionName, d.updateWritableConfig); err != nil {
+		return errors.NewCommonEdgeX(errors.Kind(err), fmt.Sprintf("unable to listen for changes for '%s' custom configuration", InsecureSecretsConfigSectionName), err)
+	}
+	// Add listener for changes in opcua custom config section
+	if err := ds.ListenForCustomConfigChanges(&d.serviceConfig.OPCUAServer.Writable, CustomConfigSectionName, d.updateWritableConfig); err != nil {
+		return errors.NewCommonEdgeX(errors.Kind(err), fmt.Sprintf("unable to listen for changes for '%s' custom configuration", CustomConfigSectionName), err)
 	}
-
 	return nil
 }
 
+// GetEndpoints capsules the containing method for easy mocking in unit tests
 var (
 	GetEndpoints = opcua.GetEndpoints
 )
 
+// SelectEndPoint capsules the containing method for easy mocking in unit tests
 var (
 	SelectEndPoint = opcua.SelectEndpoint
 )
 
+// ReadCertAndKey capsules the containing method for easy mocking in unit tests
+var (
+	ReadCertAndKey = ReadClientCertAndPrivateKey
+)
+
+// CertKeyPair capsules the containing method for easy mocking in unit tests
+var (
+	CertKeyPair = tls.X509KeyPair
+)
+
+func ReadClientCertAndPrivateKey(clientCertFileName, clientKeyFileName string) ([]byte, []byte, error) {
+	clientCertificate, err := os.ReadFile(clientCertFileName)
+	var privateKey []byte = nil
+
+	if err != nil {
+		log.Println("Client certificate not existing, creating new one")
+
+		clientCert, clientKey, err := CreateSelfSignedClientCertificates("localhost")
+		if err != nil {
+			return nil, nil, err
+		}
+		err = os.WriteFile(clientCertFileName, clientCert, 0777)
```

Comment:
0777 the file read and write permissions are reusable IMHO they should be stored in a constant or variable and used, same goes for the localhost etc. I always find it hard to have the hardcoded values and would very much appreciate if they are rather used as variables etc.

Reply:
done